lib/core/src/persist/address.rs

```diff
@@ -0,0 +1,78 @@
+use anyhow::Result;
+use log::debug;
+use rusqlite::{Connection, Row};
+
+use crate::error::PaymentError;
+
+use super::{Persister, ReservedAddress};
+
+impl Persister {
+    pub(crate) fn get_expired_reserved_address(&self, tip: u32) -> Result<Option<ReservedAddress>> {
+        let con: Connection = self.get_connection()?;
+        let query = Self::get_reserved_address_query(vec!["expiry_block_height < ?1".to_string()]);
+        let res = con.query_row(&query, [tip], Self::sql_row_to_reserved_address);
+
+        Ok(res.ok())
+    }
+
+    fn get_reserved_address_query(where_clauses: Vec<String>) -> String {
+        let mut where_clause_str = String::new();
+        if !where_clauses.is_empty() {
+            where_clause_str = String::from("WHERE ");
+            where_clause_str.push_str(where_clauses.join(" AND ").as_str());
+        }
+
+        format!(
+            "
+            SELECT
+                address,
+                expiry_block_height
+            FROM reserved_addresses
+            {where_clause_str}
+            ORDER BY expiry_block_height ASC
```

Comment:
Could it be the query is missing a `LIMIT 1`?

Looks like it can return a list of reserved addresses, but in L12-13 above it's used to fetch a single row.

Reply:
It is safe to do this as other rows except the first are ignored. But I can add it for clarity
```
    /// If the query returns more than one row, all rows except the first are
    /// ignored.
```
https://github.com/rusqlite/rusqlite/blob/b7309f2dca70716fee44c85082c585b330edb073/src/lib.rs#L684-L704

Reply:
Added LIMIT 1 [70bf17b](https://github.com/breez/breez-sdk-liquid/pull/531/commits/70bf17bd8ee3833969f07da8a28df8c0fb307419)